OCPBUGS#6729: Validating admission webhook can block etcd cluster restore procedure (#56284)
openshift/openshift-docs: modules/dr-restoring-cluster-state.adoc

====
+
. Restart the Open Virtual Network (OVN) Kubernetes pods on all the hosts.
+
[NOTE]
====

> Validating webhooks may reject pods. If you add any additional validating webhooks and those webhooks reject pods, then you must disable the webhooks while restoring the cluster state. After restoring the cluster state successfully, you can enable the webhooks. If the` failurePolicy` is set to `Fail`, the process of restoring the cluster state fails. To avoid this failure, you can temporarily	set the `failurePolicy` to `Ignore`.

Reviewer: ```suggestion
Validating and mutating admission webhooks may reject pods. If you add any additional webhooks and the `failurePolicy` is set to `Fail` in those webhooks, then they may reject pods. As a result the restoration process may fail. This can be avoided in two ways. One way is to disable (save and delete) those webhooks while restoring the cluster state. After the cluster state is restored successfully, the webhooks can be enabled (recreate) again. The other way is to temporarily set the `failurePolicy` to `Ignore` if it is set to `Fail` in the webhooks while restoring the cluster state. After the cluster state is restored successfully,  `failurePolicy`  can again be set to `Fail`.
```

Author: Addressed

> Validating and mutating admission webhooks may reject pods. If you add any additional webhooks and the `failurePolicy` is set to `Fail` in those webhooks, then they may reject pods. As a result, the restoration process may fail. You can avoid this failure by disabling or saving and deleting those webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable or recreate the webhooks again. Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.

Reviewer: ```suggestion
Validating and mutating admission webhooks may reject pods. If you add any additional webhooks and the `failurePolicy` is set to `Fail` in those webhooks, then they may reject pods. As a result, the restoration process may fail. You can avoid this failure by disabling i.e. saving and deleting those webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable i.e. recreate the webhooks again. Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.
```

Author: Addressed

> Validating and mutating admission webhooks may reject pods. If you add any additional webhooks and the `failurePolicy` is set to `Fail` in those webhooks, then they may reject pods. As a result, the restoration process may fail. You can avoid this failure by disabling that is by saving and deleting those webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable that is recreate the webhooks again. Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.

Reviewer 2: IBM style says to use "may" with caution. I would suggest using "can" or "might" as the style guide suggests. https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m (see the word "may" under the M's)

Author: Addressed

> Validating and mutating admission webhooks may reject pods. If you add any additional webhooks and the `failurePolicy` is set to `Fail` in those webhooks, then they may reject pods. As a result, the restoration process may fail. You can avoid this failure by disabling that is by saving and deleting those webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable that is recreate the webhooks again. Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.

Reviewer 2: I have a couple suggestions for minimalism to consider. If it takes from the meaning, feel free to disregard.

```suggestion
Validating and mutating admission webhooks can reject pods. If you add any additional webhooks with the `failurePolicy` set to `Fail`, then they can reject pods and the restoration process can fail. You can avoid this by saving and deleting webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable the webhooks again. 

Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.
```

Author: Addressed
